walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,121 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
+  /**
+   * create a query to select available time slots based on request.
+   * @param events existing events for attendees to participate
+   * @param request request of meetings with specific time and attendees
+   * @return a collection of available time slots
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> requiredAttendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    long requiredDuration = request.getDuration();
+
+    // Check edge cases when the required duration is too long
+    if (requiredDuration > TimeRange.WHOLE_DAY.duration()) {
+      return new ArrayList<>();
+    }
+
+    // Find the available time slots for required attendees and optional attendees
+    ArrayList<TimeRange> requiredTime = new ArrayList<>(
+        Collections.singletonList(
+            TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true)
```

Comment:
Can possibly use this field?
public static final TimeRange WHOLE_DAY = new TimeRange(0, 24 * 60);
You can also combine the "));" below to this line.